autograder/handgrading/tests/test_views/test_handgrading_result_views.py

```diff
@@ -670,71 +691,107 @@ def test_get_handgrading_results_exclude_staff(self):
 
     def test_handgrader_get_paginated_handgrading_results_specific_page(self):
-        self.do_handgrading_results_test(self.handgrader, num_results=4, page_size=3, page_num=2)
+        self.do_handgrading_results_test(self.handgrader,
+                                         num_groups=4, page_size=3, page_num=2)
 
     def test_get_paginated_results_some_groups_have_no_result(self):
-        self.do_handgrading_results_test(self.handgrader, num_results=2, num_groups=4)
+        self.do_handgrading_results_test(
+            self.handgrader, num_groups=4, submission_configs=[
+                _GroupHGDataConfig()] * 2 + [_GroupHGDataConfig(has_handgrading_result=False)] * 2)
 
     def test_non_staff_non_handgrader_get_handgrading_results_permission_denied(self):
         [student] = obj_build.make_student_users(self.course, 1)
         self.client.force_authenticate(student)
         response = self.client.get(reverse('handgrading_results', kwargs={'pk': self.project.pk}))
         self.assertEqual(status.HTTP_403_FORBIDDEN, response.status_code)
 
-    def do_handgrading_results_test(self, user: User, *,
-                                    num_results: int, num_groups: int = None,
+    def test_handgrader_get_handgrading_results_only_rejected_submissions(self):
+        self.do_handgrading_results_test(
+            self.handgrader,
+            num_groups=3,
+            submission_configs=[
+                _GroupHGDataConfig()] + 2 * [
+                _GroupHGDataConfig(
+                    submission_status=Submission.GradingStatus.rejected,
+                    has_handgrading_result=False)])
+
+    def test_staff_get_handgrading_results_only_rejected_submissions(self):
+        self.do_handgrading_results_test(
+            self.staff,
+            num_groups=3,
+            submission_configs=[
+                _GroupHGDataConfig()] + [
+                _GroupHGDataConfig(
+                    submission_status=Submission.GradingStatus.rejected,
+                    has_handgrading_result=False)] * 2)
+
+    def do_handgrading_results_test(self, user: User, *, num_groups: int = None,
+                                    submission_configs:
+                                    list[_GroupHGDataConfig] | None = None,
                                     page_size: int = None, page_num: int = None):
-        if num_groups is None:
-            num_groups = num_results
+        """Perform the test for the handgrading results view."""
+        if submission_configs is None:
+            # default initialization of submission_configs to all finished grading submissions
+            submission_configs = [_GroupHGDataConfig()] * num_groups
         else:
-            assert num_groups >= num_results
+            assert len(submission_configs) == num_groups
+
+        # ensure that finished grading submissions are sorted before rejected submissions
+        submission_configs.sort(
```

Comment:
Do we need this sorting? As far as I can tell, it shouldn't matter which order the groups are in.
Maybe `submission_configs` wasn't the best suggestion for name--it should probably be something like `group_handgrading_data_configs` instead since we're only making one submission per group.

Reply:
Ahh. This was necessary from the old way of generating expected data. I will remove this!